Packs/AWS-IAMIdentityCenter/Integrations/AWSIAMIdentityCenter/AWSIAMIdentityCenter.py

```diff
@@ -2,244 +2,427 @@
+        AlternateIdentifier={
+            'UniqueAttribute': {
+                'AttributePath': "displayName",
                 'AttributeValue': f'{groupName}'
-            },
-        ]
-    )
-    rep = json.dumps(response)
-    repJSON = json.loads(rep)
-    datas = repJSON.get('Groups', [])
-    for da in datas:
-        for group in response['Groups']:
-            group_details = {
-                'DisplayName': group['DisplayName'],
-                'GroupId': group['GroupId']
             }
-            groupID = group['GroupId']
-            data.append(group_details)
-    ec = {'AWS.IAM.IdentityCenter.Groups': data}
-    human_readable = tableToMarkdown('AWS IAM Identity Center Groups', data)
-    return_outputs(human_readable, ec)
-    return groupID
-
-
-def list_groups_for_user(args, client):  # pragma: no cover
-    data = []
-    userName = demisto.getArg('userName')
-    userID = get_user(args, client)
-    response = client.list_group_memberships_for_member(
-        IdentityStoreId=f'{IDENTITYSTOREID}',
-        MemberId={
-            'UserId': f'{userID}'
         }
     )
-    rep = json.dumps(response)
-    repJSON = json.loads(rep)
-    datas = repJSON.get('GroupMemberships', [])
-    for da in datas:
-        for group in response['GroupMemberships']:
-            group_details = {
-                'GroupId': group['GroupId'],
-                'MembershipId': group['MembershipId']
-            }
-            membershipID = group['MembershipId']
-            data.append(group_details)
-    ec = {'AWS.IAM.IdentityCenter.Users.GroupMemeberships': data}
-    human_readable = tableToMarkdown(f'AWS IAM Identity Center Group for user {userName} ', data)
-    return_outputs(human_readable, ec)
-    return membershipID
+    return response_id
+
+
+def get_group(args, client, IdentityStoreId):
+    response_id = get_groupId_by_displayName(args, client, IdentityStoreId)
+    response = client.describe_group(
+        IdentityStoreId=f'{IdentityStoreId}',
+        GroupId=response_id.get('GroupId')
+    )
+    hr_data = {
+        'DisplayName': response.get('DisplayName'),
+        'GroupId': response['GroupId']
+    }
+    del response['ResponseMetadata']
+    ec = {'AWS.IAMIdentityCenter.Group(val.GroupId == obj.GroupId)': response}
+    human_readable = tableToMarkdown('AWS IAM Identity Center Groups', hr_data, removeNull=True)
+    result = CommandResults(
+        readable_output=human_readable,
+        outputs_key_field='GroupId',
+        outputs=ec
+    )
+    return_results(result)
 
 
-def add_user_to_group(args, client):  # pragma: no cover
-    userID = get_user(args, client)
-    GroupID = get_group(args, client)
+def list_groups_for_user(args, client, IdentityStoreId):
+    hr_data = []
+    context_data = {}
+    userID = get_userId_by_username(args, client, IdentityStoreId)['UserId']
+    kwargs = {
+        'IdentityStoreId': f'{IdentityStoreId}',
+        'MemberId': {
+            'UserId': f'{userID}'
```

Comment:
What will happen if get_userId will not return any user?

Reply:
It returns an error - User not found